src/query.ts

```diff
@@ -216,16 +216,22 @@ export class Query {
    *                                         (default) or an "exclude" filter.
    *
    */
-  public toggleFilter(filterName: string, value: FacetOption, filterType = 'filter'): void {
-    if (!isPlainObject(value)) {
-      (value as Array<unknown>).forEach((value: any) => {
-        if (!this.hasFilter(filterName, value, filterType)) {
-          this.addFilter(filterName, [value], filterType);
-        } else {
-          this.removeFilter(filterName, [value], filterType);
-        }
-      });
+  public toggleFilter(filterName: string, value: FacetOption | string | number, filterType = 'filter'): void {
+    let values: FacetOption = [];
+
+    if (isArray(value)) {
+      values = value as FacetOption;
+    } else if (typeof value === 'string' || typeof value === 'number') {
+      (values as Array<unknown>).push(value);
```

Comment:
You could try this:

```typescript
let values: Array<RangeFacet|string|number> = [];

if (Array.isArray(value)) {
  values = value;
} else if (isString(value) || isNumber(value) || isPlainObject(value)) {
  values = [value];
}
```

It takes into account plain objects (`RangeFacet`). It will probably work only if you pass them by reference, but I will provide you an implementation of `shallowEqual` to compare plain objects so you can compare by value inside `hasFilter`.

Reply:
Facebook guys did a great job here (no deps): https://github.com/facebook/fbjs/blob/master/packages/fbjs/src/core/shallowEqual.js

Reply:
The other possibility is that `hasFilter` returns `true` if you ask if there's a filter in a numeric field (the filter is a `RangeFacet`) so you just remove it without checking equality or set it if there's no filter.